packages/hoppscotch-common/src/services/persistence/validation-schemas/index.ts

```diff
@@ -339,12 +348,38 @@ const HoppTestDataSchema = z.lazy(() =>
     .strict()
 )
 
-const EnvironmentVariablesSchema = z
-  .object({
-    key: z.string(),
-    value: z.string(),
-  })
-  .strict()
+const EnvironmentVariablesSchema = z.union([
+  z
+    .object({
+      key: z.string(),
+      value: z.string(),
+      secret: z.literal(false),
+    })
+    .strict(),
+  z
+    .object({
+      key: z.string(),
+      secret: z.literal(true),
+    })
+    .strict(),
+])
+
+export const SecretEnvironmentVariable = z.union([
```

Comment:
```suggestion
export const SECRET_ENVIRONMENT_VARIABLE_SCHEMA = z.union([
```

Let's follow the uppercase convention used for exposed schemas.